tensorflow/python/training/checkpointable/util.py

```diff
@@ -969,6 +969,12 @@ def assert_existing_objects_matched(self):
         raise AssertionError(
             "Object not assigned a value from checkpoint: %s" % (node,))
     for checkpointable_object in list_objects(self._root_checkpointable):
+      # Remove data structures that do not contain any variables from
+      # restoration checks.
+      if (isinstance(checkpointable_object,
+                     data_structures.CheckpointableDataStructure) and
+              len(checkpointable_object.variables) == 0):
```

Comment:
`len(checkpointable_object._checkpoint_dependencies) == 0`? Or just `not checkpointable_object._checkpoint_dependencies`. So that we make sure there are no non-variable dependencies (Layers, other checkpointable objects).

Reply:
That's even better. I didn't know that existed.
Regarding the test. Is there anything under which I can include it already like `CheckpointingTests` or should I add a new case? I would then reuse part of the code in my issue and see how you perform testing.

Reply:
I don't have time to look into bazel today. I have added a test and changed the `if`clause.